docs/How_config_files_are_merged.md

```diff
@@ -0,0 +1,576 @@
+
+### **Advantages of only using a project configuration**
+
+A project configuration is simply a team configuration file that resides in an arbitrary directory of your choice. Typically, that directory would contain application source code files for one of your projects. That is why we refer to it as a project configuration.
+
+Typically, your application source files are maintained in some source control tool. A zowe configuration file located in that same directory could also be placed into source control. If so, each member of your team who retrieves the application source, will also automatically retrieve the Zowe configuration file from source control. In this scheme every member of your team can easily share the same configuration when working on that particular application project.
+
+If you work on multiple projects, by simply changing your directory into the desired project directory, your Zowe CLI configuration will automatically change to the configuration that is appropriate for that project.
+
+Whether you share a global Zowe configuration with teammates in an ad-hoc fashion, or share a project Zowe configuration with teammates through a source control tool; others on your team can easily share the same working Zowe configuration. That is why we generically refer to our Zowe CLI V2 configuration as a `team configuration`.
+
+It is when you are using a Zowe CLI configuration in a project directory that you are most likely to use a user configuration. A user configuration has the same structure as a team configuration (although you typically would only specify a small subset of properties that are specific to your needs.) For example, perhaps you need to debug a special test instance of some service, while the rest of your team might need to use a production instance of that same service. You could place the port number of your special test instance into your user config file. Everyone shares all of the other properties from the project team configuration, but only your user config file will override the port number to point to your test instance. With such a scheme, every user could connect to their own test instance of a service, by each user specifying a unique port number in their own user config file.
+
+While you commit your project team configuration file into your source control tool, you **NEVER** commit your user config file into source control. This is what enables the team to share most properties, but still have a few personal overrides when needed.
```

Comment:
In addition to this warning, should we recommend adding the user config file to an ignore list for their VCS tool (such as `.gitignore`)?

For folks who aren't super familiar with how Git works, it may not be obvious to them that the file can be added to such an ignore list. We don't want them to think that for every commit, they need to be concerned about the risk of accidentally staging the user config file in their list of changes.

Reply:
Good idea.